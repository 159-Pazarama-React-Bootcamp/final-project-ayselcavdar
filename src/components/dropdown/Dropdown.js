import PropTypes from 'prop-types';
import styles from './Dropdown.module.css';

const Dropdown = ({
  label,
  value,
  data,
  customClass,
  name,
  onChange,
  ...rest
}) => {
  return (
    <div className={`${styles['form-group--select']} ${customClass}`}>
      <select
        name={name}
        value={value}
        className="form-control"
        onChange={onChange}
        {...rest}
      >
        <option value="">{label}</option>
        {data.map((item, key) => (
          <option key={key} value={item}>
            {item}
          </option>
        ))}
      </select>
    </div>
  );
};

Dropdown.propTypes = {
  name: PropTypes.string,
  value: PropTypes.string,
  label: PropTypes.string,
  placeholder: PropTypes.string,
  data: PropTypes.array.isRequired,
  customClass: PropTypes.string,
  onChange: PropTypes.func.isRequired,
};

Dropdown.defaultProps = {
  label: '',
  value: '',
  customClass: '',
  placeholder: '',
};

export default Dropdown;
