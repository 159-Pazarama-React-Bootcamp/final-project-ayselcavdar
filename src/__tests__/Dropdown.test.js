import render from 'react-test-renderer';
import Dropdown from '../components/dropdown/Dropdown';

describe('<Dropdown /> rendering', () => {
  it('Drop down comp has only one select element', () => {
    const data = ['ONE', 'TWO'];
    const onChange = jest.fn();
    const wrapper = shallow(<Dropdown data={data} onChange={onChange} />);
    expect(wrapper.find('select')).toHaveLength(1);
  });

  it('Dropdown snapshots check', () => {
    const data = ['ONE', 'TWO'];
    const onChange = jest.fn();
    const comp = render.create(<Dropdown data={data} onChange={onChange} />);
    let tree = comp.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
