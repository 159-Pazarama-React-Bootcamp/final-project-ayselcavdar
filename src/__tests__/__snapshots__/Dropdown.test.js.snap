// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown /> rendering Dropdown render check 1`] = `
<div
  className="form-group--select "
>
  <select
    className="form-control"
    onChange={[MockFunction]}
    placeholder=""
    value=""
  >
    <option
      value=""
    >
      
    </option>
    <option
      value="ONE"
    >
      ONE
    </option>
    <option
      value="TWO"
    >
      TWO
    </option>
  </select>
</div>
`;

exports[`<Dropdown /> rendering Dropdown snapshots check 1`] = `
<div
  className="form-group--select "
>
  <select
    className="form-control"
    onChange={[MockFunction]}
    placeholder=""
    value=""
  >
    <option
      value=""
    >
      
    </option>
    <option
      value="ONE"
    >
      ONE
    </option>
    <option
      value="TWO"
    >
      TWO
    </option>
  </select>
</div>
`;

exports[`Dropdown render check 1`] = `
<div
  className="form-group--select "
>
  <select
    className="form-control"
    onChange={[MockFunction]}
    placeholder=""
    value=""
  >
    <option
      value=""
    >
      
    </option>
    <option
      value="ONE"
    >
      ONE
    </option>
    <option
      value="TWO"
    >
      TWO
    </option>
  </select>
</div>
`;
